addons/l10n_de/models/res_config_settings.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if self.env.company.country_code == "DE" and vals.get("tax_calculation_rounding_method") == 'round_globally':
```

Comment:
the main company is the default company_id in the settings, but most of the field refer to company_id (imagining that it could be changed, this could make sense) so maybe we should refer to the current record company_id rather than self.env.company

also there is the field `res.company().country_code`, but also `res.config.settings().country_code` that refers to `company_id.account_fiscal_country_id.code`. I'm not sure which would be better to use, they probably are the same most of the time but `account_fiscal_country_id` seems `readonly=False`.

Reply:
I used the `res.company().country_code` just to make sure